File: Deal/src/main/java/ru/leonov/deal/mappers/LoanRequestMapper.java
package ru.leonov.deal.mappers;

import org.mapstruct.Mapper;
import org.mapstruct.Mapping;
import org.mapstruct.factory.Mappers;
import ru.leonov.deal.dto.LoanApplicationRequestDTO;
import ru.leonov.deal.model.entity.ClientEntity;

@Mapper
public interface LoanRequestMapper {

    LoanRequestMapper INSTANCE = Mappers.getMapper(LoanRequestMapper.class);

    @Mapping(source = "passportSeries", target = "passport.series")
    @Mapping(source = "passportNumber", target = "passport.number")
    @Mapping(source = "birthdate", target = "birthDate")
    ClientEntity loanRequestToClient(LoanApplicationRequestDTO loanApplicationRequestDTO);

}

File: Deal/src/main/kotlin/ru/leonov/deal/mapper/LoanRequestMapper.kt
package ru.leonov.deal.mapper

import ru.leonov.deal.dto.LoanApplicationRequestDTO
import ru.leonov.deal.model.entity.ClientEntity
import ru.leonov.deal.model.record.PassportRecord

/**
 * Create new incomplete [ClientEntity] from [LoanApplicationRequestDTO].
 */
fun loanRequestToClient(loanApplicationRequestDTO: LoanApplicationRequestDTO): ClientEntity {
    val passportRecord = PassportRecord(
        series = loanApplicationRequestDTO.passportSeries.toInt(),
        number = loanApplicationRequestDTO.passportNumber.toInt()
    )
    return ClientEntity(
        firstName = loanApplicationRequestDTO.firstName,
        lastName = loanApplicationRequestDTO.lastName,
        middleName = loanApplicationRequestDTO.middleName,
        birthDate = loanApplicationRequestDTO.birthdate,
        email = loanApplicationRequestDTO.email,
        passport = passportRecord
    )
}
